app/utils/database.py

```diff
@@ -296,6 +353,48 @@ def frontend_data(self):
                 'validated': self.validated if self.validated else False,
                 }
 
+    def genebank_permission(self, genebank_id):
+        """
+        Returns the permission level for the genebank given by `genebank_id`.
+        Permission levels are set as 'public', 'authenticated', 'private'.
+        """
+
+        access_level = 'public'
+        for role in self.privileges['roles']:
+            if role['role'] == 'admin':
```

Comment:
The way you are using the name role now is for two things:
```typescript
type role₁ = {role₂: role₂, target: string]
type role₂ = 'admin' | 'specialist' | 'manager' | 'owner'
```
In addition, privileges are only roles:
```typsecript
type privileges = {roles₁: role₁[]}
```
This dual naming is apparent in this code which with the indexes would be
```python
for role₁ in self.privileges['roles₁']:
    if role₁['role₂'] == 'admin':
         ...
```
I don't know what you're after but one way to simplify this assuming that there are no other privs than roles would be:
```typescript
type privileges = privilege[]
type privilege = { role: role, target: string }
type role = 'admin' | 'specialist' | 'manager' | 'owner'
```
A less invasive change would be to change `role₁` to `targeted_role`:
```typescript
type privileges = {targeted_roles: targeted_role[]}
type targeted_role = {role: role, target: string]
type role = 'admin' | 'specialist' | 'manager' | 'owner'
```